framework/src/functions/MooseParsedFunction.C

```diff
@@ -71,6 +72,21 @@ template <typename T>
 void
 MooseParsedFunctionTempl<T>::initialSetup()
 {
+  for (auto i = 0; i < (int)_vars.size(); i++)
```

Comment:
you dont need to cast if you have auto

Reply:
Even better:
`for (const auto i : index_range(_vars))`

Reply:
wasn't familiar with this index_range function! very useful.